test/SlowTests/MailingList/IndexCompilation.cs

```diff
@@ -222,7 +224,7 @@ public void CanCompileIndexWithDistinct()
                             SelectSum = listOfUsers[i].LoginCountByDate.ToDictionary(y => y.Key, y => y.Value).OrderBy(x => x.Value).Select(x => x.Value).Sum(x => x),
                             OrderBySum = listOfUsers[i].LoginCountByDate.ToDictionary(y => y.Key, y => y.Value).Select(x => x.Value).Sum(x => x),
                             IdsWithDecimals = listOfUsers[i].ListOfDecimals.ToDictionary(k => k, k => 0),
-                            OrderByDescending = listOfUsers[i].LoginCountByDate.OrderByDescending(x => x.Value).ToDictionary(y => y.Key.ToString(CultureInfo.InvariantCulture), y => y.Value),
+                            OrderByDescending = listOfUsers[i].LoginCountByDate.OrderByDescending(x => x.Value).ToDictionary(y => y.Key.GetDefaultRavenFormat(isUtc: y.Key.Kind == DateTimeKind.Utc), y => y.Value),
```

Comment:
Why you need to change it? Would it be a breaking change?

Reply:
this is how we store dicitonary key if it is DateTime

Reply:
Does it mean that if someone has the index definition as in this test (with the usage of `y.Key.ToString(CultureInfo.InvariantCulture)` then his index will stop working after merging this PR? That would be breaking change - that's my concern

Reply:
If he has a DateTime dicitonary key then his index doesnt work now, because in document we store it using GetDefaultRavenFormat
and in dictionary we store it just as ToString
https://github.com/ravendb/ravendb/blob/v4.2/src/Raven.Server/Utils/TypeConverter.cs#L243

Reply:
Okay, I see this test was added by you in your previous PR related to the same issue (https://github.com/ravendb/ravendb/pull/9351/files#diff-2fb714770e61aff7a2f4afd3c71eb209R225)

That's fine then. Just wanted to double check that.